tests/integration/test_merge_tree_s3/test.py

```diff
@@ -582,7 +578,7 @@ def test_move_replace_partition_to_another_table(cluster, node_name):
         cluster,
         FILES_OVERHEAD * 2
         + FILES_OVERHEAD_PER_PART_WIDE * 4
-        - FILES_OVERHEAD_METADATA_VERSION * 2,
+        + FILES_OVERHEAD_METADATA_VERSION * 2,
```

Comment:
In my understanding this `+ FILES_OVERHEAD_METADATA_VERSION *` is necessary because we use the objects from the other table, but we write our own metadata version. This is necessary, because the two metadata version are not connected.

Reply:
Yes, but I guess we should just remove `- FILES_OVERHEAD_METADATA_VERSION * 2` instead of replacing it with `+ FILES_OVERHEAD_METADATA_VERSION * 2` because the metadata version files are already counted in `FILES_OVERHEAD_PER_PART_WIDE`, we just need to not subtract it

Reply:
I think we have to count, because what is happening is:
1. We [use hard links](https://github.com/ClickHouse/ClickHouse/blob/17fda33a042d428df3363a50a03a3fb545fc7a29/src/Storages/StorageMergeTree.cpp#L2132-L2135) instead of copy (`ClonePartParams::copy_instead_of_hardlink` is by default `false`), thus the number of objects shouldn't change because of the hardlinks
2. But we write two new metadata version file. That's why the extra `FILES_OVERHEAD_METADATA_VERSION * 2`.

I can try to check it manually what happens, if you think this strange or shouldn't happen. In my mind this makes sense.